# Model Inference

## Import Libraries

In [1]:
# Libraries for Data Manipulation
import pandas as pd
# Libraries for saving
import pickle
import json

## Load Model

In [3]:
# Load model and other files
with open('model_knn.pkl', 'rb') as file_1:
  model_knn_inf = pickle.load(file_1)

# Load list kolom numerikal
with open('list_num_cols.txt', 'r') as file_2:
  num_cols_inf = json.load(file_2)

# Load list kolom kategorikal from JSON file
with open('list_cat_cols.txt', 'r') as file_3:
  cat_cols_inf = json.load(file_3)

In [4]:
# Data dibuat dengan random  dan dijadikan dataframe.

test = [{'BHK' : 2,
        'Rent': 15000,
        'Size' : 750, 
        'Floor' : 1, 
        'Area Type' : 'Super Area', 
        'Area Locality' : 'Dumdum Park', 
        'City' : 'Koltaka',
        'Furnishing Status' : 'Semi-Furnished', 
        'Tenant Preferred' : 'Bachelors', 
        'Bathroom' : 1,
        'Point of Contact' : 'Contact Owner'
}]

test_inference = pd.DataFrame(test)

# Tampilkan DataFrame
test_inference

BHK   Rent  Size  Floor   Area Type Area Locality     City  \
0    2  15000   750      1  Super Area   Dumdum Park  Koltaka   

  Furnishing Status Tenant Preferred  Bathroom Point of Contact  
0    Semi-Furnished        Bachelors         1    Contact Owner

In [5]:
# Transform the new data using the pipeline
predictions = model_knn_inf.predict(test_inference)

# Print the predictions
print("Predictions Harga Sewa:", predictions)


Predictions Harga Sewa: [11571.42857143]


**Insight** : Hasil prediksi menunjukkan bahwa harga sewa yang diprediksi oleh model KNN adalah sebesar 11571.42857143. Dengan demikian, insight yang dapat diambil dari hasil ini adalah prediksi harga sewa untuk properti yang diberikan oleh model KNN.